src/software/new_geom/angle.h

```diff
@@ -0,0 +1,595 @@
+#pragma once
+
+#include <cmath>
+#include <ostream>
+
+#include "software/new_geom/geom_constants.h"
+
+/**
+ * A typesafe representation of an angle.
+ *
+ * This class helps prevent accidentally combining values in degrees and radians
+ * without proper conversion.
+ */
+class Angle final
+{
+   public:
+    /**
+     * The zero angle.
+     */
+    static constexpr Angle zero();
+
+    /**
+     * The quarter-turn angle (90°).
+     */
+    static constexpr Angle quarter();
+
+    /**
+     * The half-turn angle (180°).
+     */
+    static constexpr Angle half();
+
+    /**
+     * The three-quarter turn angle (270°).
+     */
+    static constexpr Angle threeQuarter();
+
+    /**
+     * The full-turn angle (360°).
+     */
+    static constexpr Angle full();
+
+    /**
+     * Constructs an angle from a value in radians.
+     *
+     * @param rad the angle in radians.
+     *
+     * @return the constructed angle
+     */
+    static constexpr Angle ofRadians(double rad);
```

Comment:
I'd prefer fromRadians

```suggestion
    static constexpr Angle fromRadians(double rad);
```

Reply:
Agreed, this has annoyed me for a while.